### PR TITLE
TASK-55998: fix dates issues

### PR DESCRIPTION
**Pull request by GouadriaHanen**
fixed issues by changing date.toInstant() by LocalDateTime when parsing in/To RFC3339Date since :
The LocalDateTime represents date-time without a time-zone such as 2019-10-25T12:15:30 whereas Instant is an instantaneous point on the time-line

### CODE REVIEW

File: services/src/main/java/org/exoplatform/addons/gamification/utils/Utils.java [hunk 1]
```diff
@@ -114,16 +115,16 @@ public static String toRFC3339Date(Date dateTime) {
...
   }
 
   public static Date parseRFC3339Date(String dateString) {
     if (StringUtils.isBlank(dateString)) {
       return null;
     }
-    ZonedDateTime zonedDateTime = ZonedDateTime.parse(dateString, RFC_3339_FORMATTER);
-    return Date.from(zonedDateTime.toInstant());
+    ZonedDateTime zonedDateTime = LocalDateTime.parse(dateString, RFC_3339_FORMATTER).atZone(ZoneId.systemDefault()).withZoneSameLocal(ZoneOffset.UTC);
+    return new Date(Timestamp.valueOf(zonedDateTime.toLocalDateTime()).getTime());
```
Comment (boubaker) [inline services/src/main/java/org/exoplatform/addons/gamification/utils/Utils.java]:
```suggestion
    LocalDateTime localDateTime = LocalDateTime.parse(dateString, RFC_3339_FORMATTER);
    return Date.from(localDateTime.toInstant(ZoneOffset.UTC));
```

File: services/src/main/java/org/exoplatform/addons/gamification/utils/Utils.java [hunk 2]
```diff
@@ -114,16 +115,16 @@ public static String toRFC3339Date(Date dateTime) {
     if (dateTime == null) {
       return null;
     }
-    ZonedDateTime zonedDateTime = ZonedDateTime.from(dateTime.toInstant().atOffset(ZoneOffset.UTC));
+    ZonedDateTime zonedDateTime = dateTime.toInstant().atZone(ZoneOffset.systemDefault()).withZoneSameLocal(ZoneOffset.UTC);
```
Comment (boubaker) [inline services/src/main/java/org/exoplatform/addons/gamification/utils/Utils.java]:
```suggestion
    ZonedDateTime zonedDateTime = dateTime.toInstant().atZone(ZoneOffset.UTC)
```

Comment (GouadriaHanen) [reply services/src/main/java/org/exoplatform/addons/gamification/utils/Utils.java]: done!

File: services/src/main/java/org/exoplatform/addons/gamification/utils/Utils.java (hunk 1 above)
Comment (GouadriaHanen) [reply services/src/main/java/org/exoplatform/addons/gamification/utils/Utils.java]: done!

**Review by boubaker — APPROVED** [review]

**Pull request merged**